core/blockchain_impl.go

```diff
@@ -1407,7 +1402,7 @@ func (bc *BlockChainImpl) WriteBlockWithState(
 		triedb.Reference(root, common.Hash{}) // metadata reference to keep trie alive
 		bc.triegc.Push(root, -int64(block.NumberU64()))
 
-		if current := block.NumberU64(); current > triesInMemory {
+		if current := block.NumberU64(); current > bc.cacheConfig.TriesInMemory {
```

Comment:
This condition is always `true`. Is it what we want?

Reply:
Later we take 
`header := bc.GetHeaderByNumber(current - triesInMemory)`
so this check seems reasonable.